src/rand.rs

```diff
@@ -119,14 +119,20 @@ use self::sysrand_or_urandom::fill as fill_impl;
 
 #[cfg(any(target_os = "linux", windows))]
 mod sysrand {
-    use {bssl, error};
+    use error;
 
     pub fn fill(dest: &mut [u8]) -> Result<(), error::Unspecified> {
-        let chunk_len = unsafe { super::GFp_sysrand_chunk_max_len };
-        for mut chunk in dest.chunks_mut(chunk_len) {
-            try!(bssl::map_result(unsafe {
-                super::GFp_sysrand_chunk(chunk.as_mut_ptr(), chunk.len())
-            }));
+        let mut read_len = 0;
+        while read_len < dest.len() {
+            let r = unsafe {
+                super::GFp_sysrand_chunk(dest[read_len ..].as_mut_ptr(), dest.len() - read_len)
+            };
+            if r < 0 {
+                return Err(error::Unspecified);
+            }
+            // XXX: If r == 0 then this is a busy wait loop waiting for the
+            //      kernel entropy pool to be initialized
```

Comment:
NIT: Don't indent the second line of the comment by the extra width of "XXX:",
NIT: End the sentence with a period.